node/node.go

```diff
@@ -260,12 +260,16 @@ func (n *Node) currentRole() api.NodeRole {
 func (n *Node) run(ctx context.Context) (err error) {
```

Comment:
Wondering if adding a summary here of all that happens within this function would be fruitful. I'll let you decide though.